### DISCUSSION TITLE
How to change the rke2 snapshotter to ZFS

### DISCUSSION THREAD

**Discussion by laralar** [rancher/rke2#7387] [Q&A]
in kubeadm deployed cluster, I managed to deploy containerd changing these values in /etc/containerd/config.toml

```
[plugins."io.containerd.grpc.v1.cri".containerd]
  snapshotter = "zfs"

[[plugins."io.containerd.transfer.v1.local".unpack_config]]
  snapshotter = "zfs"

```

However, when writing these in /var/lib/rancher/rke2/agent/etc/containerd/config.toml.tmpl it seems they are not applied

```
{{ template "base" . }}

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes."custom"]
  runtime_type = "io.containerd.runc.v2"
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes."custom".options]
  BinaryName = "/usr/bin/custom-container-runtime"

[plugins."io.containerd.grpc.v1.cri".containerd]
  snapshotter = "zfs"

[[plugins."io.containerd.transfer.v1.local".unpack_config]]
  snapshotter = "zfs"
```

I also tried copying the exact file from /etc/containerd.toml in the template file with no success, even after restarting rke2-server and agent.

Any help would be appreciated.  I am trying to avoid mounting a zfs/ext4 volume in  /var/lib/rancher/rke2/agent/etc/containerd/

**Comment by brandond** [comment]
https://docs.rke2.io/reference/linux_agent_config?_highlight=snapshotter#runtime

In your config.yaml: 
```yaml
snapshotter: zfs
```

Why do you need the zfs snapshotter? overlayfs should work fine on recent versions of zfs.

**Reply by laralar** [reply]
Still while deploying latest RKE2 over a ZFS filesystem I got a bunch of overlays snapshotter errors.  The only way to solve this was to mount a zfs volume block volume formatted as ext4 in the /var/lib/rancher/k3s/agent/containerd

"Failed to retrieve agent config: "overlayfs" snapshotter cannot be enabled for "/var/lib/rancher/rke2/agent/containerd", try using "fuse-overlayfs" or "native": failed to mount overlay: invalid argument"

I didn't try the snapshotted: zfs in the config.yaml. I will give it a try.

Thanks

**Reply by laralar** [reply]
Any idea if this can be specified in the rancher Web UI ?
